### ISSUE TITLE
dnsdist Debian package cannot bind to IPv4 and IPv6

### ISSUE THREAD

**Issue by frootmig** [PowerDNS/pdns#2542]
I wanted to bind dnsdist to all public IPv4 and IPv6 addresses. I configured useLocal in the configuration file. /etc/default/dnsdist contains OPTIONS='-l 127.0.0.1'

This overrides the useLocal from configuration and binds dnsdist to localhost IPv4 only.

Changing OPTIONS to '' or "" leads to the following error message:

```
root@nameserver:/# systemctl status dnsdist
● dnsdist.service - DNSdist
   Loaded: loaded (/lib/systemd/system/dnsdist.service; enabled)
   Active: inactive (dead) since Sat 2015-05-23 08:25:13 EDT; 8h ago
  Process: 11072 ExecStart=/usr/bin/dnsdist --supervised ${OPTIONS} (code=exited, status=0/SUCCESS)
 Main PID: 11072 (code=exited, status=0/SUCCESS)

May 23 08:25:13 nameserver dnsdist[11072]: Running in the foreground
May 23 08:25:13 nameserver dnsdist[11072]: Accepting control connections on 0.0.0.0:5199
May 23 08:25:13 nameserver dnsdist[11072]: Read configuration from '/etc/dnsdist/dnsdist.conf'
May 23 08:25:13 nameserver dnsdist[11072]: Added downstream server X.X.X.X:53
May 23 08:25:13 nameserver dnsdist[11072]: Added downstream server X.X.X.X:53
May 23 08:25:13 nameserver dnsdist[11072]: Added downstream server X.X.X.X:53
May 23 08:25:13 nameserver dnsdist[11072]: Running in the foreground
May 23 08:25:13 nameserver dnsdist[11072]: Accepting control connections on 0.0.0.0:5199
May 23 08:25:13 nameserver dnsdist[11072]: Webserver launched on X.X.X.X:8083
May 23 08:25:13 nameserver dnsdist[11072]: Fatal pdns error: Unable to convert presentation address ''
```

It is also not possible to set OPTIONS to '-l 127.0.0.1 -l [::]'

Everything after the first IP is then mentioned in the error message as presentation address.

I'm using the Debian 8 / Jessie package from the build daemon on Debian 8 with systemd.

Workaround: remove {$OPTIONS} in call in lib/systemd/system/dnsdist.service

**Comment by ahupowerdns** [comment]
We just committed a3d2e5b67b9c6b0a118c7c5cf26fa48b89ddf139 which may solve your issue! Packages now building.

**Comment by frootmig** [comment]
The build failed with upload failed?

**Comment by ahupowerdns** [comment]
ehm ok - that is bad luck, will see if @pieterlexis can kick it one day. Sorry!

**Comment by frootmig** [comment]
Okay, I'll check later for updated packages.

I can confirm that the manual applied patch fixes the problem.

**Comment by pieterlexis** [comment]
ah yes, there's a small change we need in master for this: https://github.com/pieterlexis/pdns/commit/b1861af1d24a8a621b6056da0705e8aa09a3c4b2 will PR soon :) (something something BBQ weekend)

**Comment by pieterlexis** [comment]
>  Workaround: remove {$OPTIONS} in call in lib/systemd/system/dnsdist.service

To be honest, the 'correct' systemd way is creating the file `/etc/systemd/system/dnsdist.service` with the following content:

```
[Service]
ExecStart=/usr/bin/dnsdist --supervised
```

Or just `rm`'ing the `/etc/default/dnsdist` file.

But I agree that emptying the var should work.

**Comment by pieterlexis** [comment]
anyway, try the packeged from (this)[http://buildmaster1.test.powerdns.plexis.eu/#/builders/7/builds/136] build and let us know if you problem is solved.

**Comment by frootmig** [comment]
I'm sorry that it took so long to verify.

The issue seems not to be fixed.

I purged the package and used the most recent build.

**Comment by frootmig** [comment]
It did not help to comment out OPTIONS in /etc/default/dnsdist

The tip with creating the file /etc/systemd/system/dnsdist.service worked as expected.

I'm not that familiar with systemd, is that something that should be done by the DEB or is this something the system administrator has to do?

**Comment by pieterlexis** [comment]
@frootmig 

> It did not help to comment out OPTIONS in /etc/default/dnsdist

:disappointed: 

> The tip with creating the file /etc/systemd/system/dnsdist.service worked as expected.

Good, this was expected.

> I'm not that familiar with systemd, is that something that should be done by the DEB or is this something the system administrator has to do?

This is something that the admin has too do. The design and philosofy of systemd doesn't really allow for the use of dfaults-files. See [here](http://0pointer.de/blog/projects/on-etc-sysinit.html). It allows this because of legacy reasons. But creating an override-service file in `/etc/systemd/system` is preferred.

I think I'll keep this ticket open and will change the systemd-based packages for dnsdist to disallow the use of defaults files.

**Comment by pieterlexis** [comment]
fixed in #3016
